import-export-cli/integration/api_test.go

```diff
@@ -63,3 +71,362 @@ func TestExportImportApiAdminSuperTenantUser(t *testing.T) {
+
+	// Import the API to env2 as tenant admin across domains
+	validateAPIImport(t, args)
+}
+
+// Export an API from one environment as super tenant admin and import to another environment as cross tenant admin
+// (without preserve-provider=false)
+func TestExportImportApiCrossTenantUser(t *testing.T) {
+	superTenantAdminUsername := superAdminUser
+	superTenantAdminPassword := superAdminPassword
+
+	superTenantApiCreator := creator.UserName
+	superTenantApiCreatorPassword := creator.Password
+
+	tenantAdminUsername := superAdminUser + "@" + TENANT1
+	tenantAdminPassword := superAdminPassword
+
+	dev := apimClients[0]
+	prod := apimClients[1]
+
+	api := addAPI(t, dev, superTenantApiCreator, superTenantApiCreatorPassword)
+
+	args := &apiImportExportTestArgs{
+		apiProvider: credentials{username: superTenantApiCreator, password: superTenantApiCreatorPassword},
+		ctlUser:     credentials{username: superTenantAdminUsername, password: superTenantAdminPassword},
+		api:         api,
+		srcAPIM:     dev,
+		destAPIM:    prod,
+	}
+
+	validateAPIExport(t, args)
+
+	// Since --preserve-provider=false is not specified, the apiProvider remain as it is and the ctlUser is tenant admin
+	args.ctlUser = credentials{username: tenantAdminUsername, password: tenantAdminPassword}
+
+	// Import the API to env2 as tenant admin across domains
+	validateAPIImportFailure(t, args)
+}
+
+func TestListApisAdminSuperTenantUser(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of API (same copy)
+	randomNumber := rand.Intn(10)
+	for apiCount := 0; apiCount <= randomNumber; apiCount++ {
+		// Add the API to env1
+		addAPI(t, dev, apiCreator, apiCreatorPassword)
+	}
+
+	args := &apiImportExportTestArgs{
+		ctlUser: credentials{username: adminUsername, password: adminPassword},
+		srcAPIM: dev,
+	}
+
+	validateAPIsList(t, args)
+}
+
+func TestListApisAdminTenantUser(t *testing.T) {
+	tenantAdminUsername := superAdminUser + "@" + TENANT1
+	tenantAdminPassword := superAdminPassword
+
+	apiCreator := creator.UserName + "@" + TENANT1
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of API (same copy)
+	randomNumber := rand.Intn(10)
+	for apiCount := 0; apiCount <= randomNumber; apiCount++ {
+		// Add the API to env1
+		addAPI(t, dev, apiCreator, apiCreatorPassword)
+	}
+
+	args := &apiImportExportTestArgs{
+		ctlUser: credentials{username: tenantAdminUsername, password: tenantAdminPassword},
+		srcAPIM: dev,
+	}
+
+	validateAPIsList(t, args)
+}
+
+func TestDeleteApiAdminSuperTenantUser(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of APIs (same copy)
+	randomNumber := rand.Intn(10)
```

Comment:
I think it is better to use a fixed number rather than a random number in this case to determine the number of APIs that will be created. A random number could lead to unpredictability that may cause unexpected failures later on.

Reply:
Sure will use a constant